File: src/main/java/com/youtube/tracker/feign/YoutubeTrackerFeign.java
package com.youtube.tracker.feign;

import com.youtube.tracker.models.YoutubeSearchResponse;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;

@FeignClient(value = "youtube-feign", url = "${youtube-api.base-url}")
public interface YoutubeTrackerFeign {

    @GetMapping("/v3/search")
    ResponseEntity<YoutubeSearchResponse> getChannels(@RequestParam String part, @RequestParam String channelId, @RequestParam String key, @RequestParam String q);
}

File: src/main/kotlin/com/youtube/tracker/feign/YoutubeTrackerFeign.kt
package com.youtube.tracker.feign

import com.youtube.tracker.enums.PartType
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.http.ResponseEntity
import com.youtube.tracker.models.YoutubeSearchResponse

@FeignClient(value = "youtube-feign", url = "\${youtube-api.base-url}")
interface YoutubeTrackerFeign {
    @GetMapping("/v3/search")
    fun search(
        @RequestParam("q") keyword: String, @RequestParam channelId: String,
        @RequestParam part: String, @RequestParam key: String, @RequestParam publishedAfter: String
    ): ResponseEntity<YoutubeSearchResponse>
}